api/grpcserver/admin_service.go

```diff
@@ -101,3 +102,28 @@ func (a AdminService) Recover(ctx context.Context, req *pb.RecoverRequest) (*emp
 	}
 	return &empty.Empty{}, nil
 }
+
+func (a AdminService) EventsStream(req *pb.EventStreamRequest, stream pb.AdminService_EventsStreamServer) error {
+	sub, err := events.Subscribe[events.UserEvent]()
+	if err != nil {
+		return status.Errorf(codes.FailedPrecondition, err.Error())
+	}
+	defer sub.Close()
+	// send empty header after subscribing to the channel.
+	// this is optional but allows subscriber to wait until stream is fully initialized.
+	if err := stream.SendHeader(metadata.MD{}); err != nil {
+		return status.Errorf(codes.Unavailable, "can't send header")
+	}
+	for {
+		select {
+		case <-stream.Context().Done():
+			return nil
+		case <-sub.Full():
```

Comment:
It would be nice to send all the buffered events at this point. Otherwise I think they'll just get lost